### DISCUSSION TITLE
Add a client-side "warn me about this player" system.

### DISCUSSION THREAD

**Discussion by civilCornball** [FakeFishGames/Barotrauma#5986] [Ideas and suggestions]
Fact: Barotrauma servers often have griefers in them. Some, however, are especially odious, and should be avoided for as long as possible, as they heavily decrease the enjoyment of the game for everyone involved.

However, those griefers often change their names or are simply forgotten by normal players. Thus, server admins are left unwarned about these harbingers of chaos.
(yes, i know, i'm being overdramatic. But I still feel that it's appropriate. Twice, the player "Jeffrey Epstein" has caused me to become banned from a server, as he has stolen my ID card and then killed several players. And I can do nothing to stop him from beyond the grave!)

To fix this problem, I propose a system that is similar to a banlist, but stored on the user side.
Via the right-click menu, players should have the option to tell the game to "warn me about this player."
Afterwards, players will be warned about that player's presence in a server they are joining. In addition, they will be warned when that player joins the game.

tl;dr:
some griefers are especially bad, and I'd like to be warned about them if I find them in other servers, so I can immediately leave.

**Comment by Daimonin** [comment]
The real problem here is that the ban system does not play well with the ID theft system.

I love the ID theft system, but when paired with server admins kicking players by ID... well it causes the above problem. Ideally, there should be some way for a server admin/host, to bypass the ID theft system in the case of initiating a ban.

Maybe something as simple as the ban process checking the ban target, and warning the host/admin if the target is not in possession of their ID card, and tell them who DOES have that persons ID card?
Maybe a way for a server admin/host to check a log of who has killed/injected who, that displays real names, not ID card names?

**Reply by ghost** [reply]
Maybe just a warning before banning them that says:

There is someone disguised as this player! You may be banning the wrong player:
[Disguised player's real username]

Also, the logs display the real name, not the disguise.
https://imgur.com/a/KnZLAdr

**Reply by James-E-A** [reply]
Bans are one thing (but the logs handle that), for server-admins.

But _players_ might want to keep their own tabs on other accounts in-game to know if it's worth bailing on a match or just staying away from certain players on the sub, and it's tedious to click through to 12 Steam profiles every match to check for Blocked players.

**Reply by StruggleDuck** [reply]
I think that one of the easiest possible fixes for this issues is to do one of 2 things
1) Make it so that you can see a disguised persons real name beside their disguise name when spectating, or 
2) Make an option for admins to enable which enables them to see a players real name.

The other thing I think that should be added is that you should be able to see that a player was disguised as someone else when they do something in the logs, so that in the event that someone says x did something, It will be easier to figure out the actual perpetrator because their name would also show the player disguised as them, and the fact that they were pretending to be x.

**Comment by Ek-Videogames** [comment]
> as he has stolen my ID card and then killed several players. And I can do nothing to stop him from beyond the grave!)

I don't see how the 'global banlist' argument will not cause **you** to have a warning on your account in this case. Seeing as it doesn't address the inherent false-positive problem, you will get blamed for it still and then now you are the one with a flagged account.

` Hot take: when will we be allowed to ban admins for being terrible and abusive.` 😆
